UpmPackageVersionEx class throws several error messages

**Describe the bug**

`UpmPackageVersionEx` class throws several error messages in Unity versions `2021.3.21f1`, `2022.2.10f1`, `2023.1.0b7`, or `2023.2.0a5` when UpmGitExtension package is installed. The error message shows that `UpmPackageVersionEx` is unable to convert the boolean type to `UnityEditor.PackageManager.UI.Internal.RegistryType`.

**To Reproduce**

1. Create a new project in Unity versions `2021.3.21f1`, `2022.2.10f1`, `2023.1.0b7`, or `2023.2.0a5`.
2. Install `UpmGitExtension` package.
3. Unity console shown as follows:

```log
Library\PackageCache\com.coffee.upm-git-extension@2.0.2\Editor\Coffee.UpmGitExtension\Extensions\UpmPackageVersionEx.cs(42,46): error CS1503: Argument 3: cannot convert from 'bool' to 'UnityEditor.PackageManager.UI.Internal.RegistryType'
Library\PackageCache\com.coffee.upm-git-extension@2.0.2\Editor\Coffee.UpmGitExtension\Extensions\UpmPackageVersionEx.cs(51,68): error CS1503: Argument 5: cannot convert from 'bool' to 'UnityEditor.PackageManager.UI.Internal.RegistryType'
Library\PackageCache\com.coffee.upm-git-extension@2.0.2\Editor\Coffee.UpmGitExtension\Extensions\UpmPackageVersionEx.cs(60,96): error CS1061: 'UpmPackageVersion' does not contain a definition for 'isUnityPackage' and no accessible extension method 'isUnityPackage' accepting a first argument of type 'UpmPackageVersion' could be found (are you missing a using directive or an assembly reference?)
```

**Expected behavior**

No error messages should appear after installing `UpmGitExtension` package in the Unity versions mentioned above.

**Environment:**

- UpmGitExtension version: 2.0.2
- Platform: Editor(Windows/Mac)
- Unity version: 
  - 2021.3.21f1
  - 2022.2.10f1
  - 2023.1.0b7
  - 2023.2.0a5

**Additional context**

The issue seems to stem from a change made to the `UpmPackageVersion` class in the Unity `PackageManagerUI` module. Here are the relevant lines in the `UnityCsReference` repository:

https://github.com/Unity-Technologies/UnityCsReference/blob/51ea2ccbe18ad3c6328f4d43b00dc74eef3be5ff/Modules/PackageManagerUI/Editor/Services/Upm/UpmPackageVersion.cs#L106

https://github.com/Unity-Technologies/UnityCsReference/blob/51ea2ccbe18ad3c6328f4d43b00dc74eef3be5ff/Modules/PackageManagerUI/Editor/Services/Upm/UpmPackageVersion.cs#L116

--- a/Editor/Coffee.UpmGitExtension/Extensions/UpmPackageVersionEx.cs
+++ b/Editor/Coffee.UpmGitExtension/Extensions/UpmPackageVersionEx.cs
@@ -10,7 +10,10 @@
 #if UNITY_2020_1_0 || UNITY_2020_1_1 || UNITY_2020_1_2 || UNITY_2020_1_3 || UNITY_2020_1_4
 #define CONSTRACTOR_V1
 #endif
-    
+#if UNITY_2021_3_21 || UNITY_2022_2_10 || UNITY_2023_1_0 || UNITY_2023_2_0
+#define REGISTRY_TYPE_HAS_BEEN_ADDED
+#endif
+
 using System;
 using System.Text.RegularExpressions;
 using UnityEngine;
@@ -34,21 +37,33 @@ namespace Coffee.UpmGitExtension
         private UnityEditor.PackageManager.PackageInfo m_PackageInfo;
         public UnityEditor.PackageManager.PackageInfo packageInfo => m_PackageInfo;
 #endif
-        
+
+#if !REGISTRY_TYPE_HAS_BEEN_ADDED
         public UpmPackageVersionEx(UnityEditor.PackageManager.PackageInfo packageInfo, bool isInstalled, bool isUnityPackage)
+#else
+        public UpmPackageVersionEx(UnityEditor.PackageManager.PackageInfo packageInfo, bool isInstalled, RegistryType availableRegistry)
+#endif
 #if CONSTRACTOR_V1
             : base(packageInfo, isInstalled)
-#else
+#elif !REGISTRY_TYPE_HAS_BEEN_ADDED
             : base(packageInfo, isInstalled, isUnityPackage)
+#else
+            : base(packageInfo, isInstalled, availableRegistry)
 #endif
         {
         }
 
+#if !REGISTRY_TYPE_HAS_BEEN_ADDED
         public UpmPackageVersionEx(UnityEditor.PackageManager.PackageInfo packageInfo, bool isInstalled, SemVersion? version, string displayName, bool isUnityPackage)
+#else
+        public UpmPackageVersionEx(UnityEditor.PackageManager.PackageInfo packageInfo, bool isInstalled, SemVersion? version, string displayName, RegistryType availableRegistry)
+#endif
 #if CONSTRACTOR_V1
             : base(packageInfo, isInstalled, version, displayName)
-#else
+#elif !REGISTRY_TYPE_HAS_BEEN_ADDED
             : base(packageInfo, isInstalled, version, displayName, isUnityPackage)
+#else
+            : base(packageInfo, isInstalled, version, displayName, availableRegistry)
 #endif
         {
         }
@@ -56,8 +71,10 @@ namespace Coffee.UpmGitExtension
         public UpmPackageVersionEx(UpmPackageVersion packageVersion)
 #if CONSTRACTOR_V1
             : base(packageVersion.GetPackageInfo(), packageVersion.isInstalled)
-#else
+#elif !REGISTRY_TYPE_HAS_BEEN_ADDED
             : base(packageVersion.GetPackageInfo(), packageVersion.isInstalled, packageVersion.isUnityPackage)
+#else
+            : base(packageVersion.GetPackageInfo(), packageVersion.isInstalled, packageVersion.availableRegistry)
 #endif
         {
 #if PACKAGE_INFO_HAS_BEEN_REMOVED

